lib/remote/deleteobjecthandler.cpp

```diff
@@ -66,6 +67,25 @@ bool DeleteObjectHandler::HandleRequest(
 	bool cascade = HttpUtility::GetLastParameter(params, "cascade");
 	bool verbose = HttpUtility::GetLastParameter(params, "verbose");
 
+	ConfigObjectsSharedLock lock (std::try_to_lock);
+
+	if (!lock) {
+		Dictionary::Ptr result1 = new Dictionary();
+
+		Dictionary::Ptr result = new Dictionary({
+			{ "results", new Array({ result1 }) }
+		});
+
+		result1->Set("errors", new Array({"Icinga is reloading"}));
+		result1->Set("code", 503);
+		result1->Set("status", "Objects could not be deleted.");
+
+		response.result(http::status::service_unavailable);
+		HttpUtility::SendJsonBody(response, params, result);
```

Comment:
Why build this complicated error message here instead of just using `HttpUtility::SendJsonError()`? Yes, the response is structured differently (simpler than what's generated here) but all three `*ObjectHandler::HandleRequest()` already use this for some errors, so that should be nothing new for clients.

Reply:
I did it just like in the rest of both the file and lib/remote/.

Reply:
In general, each case returning a custom error JSON should have a reason for doing so instead of simply using `HttpUtility::SendJsonError()`, something like there might be different errors for different objects or multiple error messages like stated in the following comment:

https://github.com/Icinga/icinga2/blob/e2fda149e536328fa7bbcb62b54725ef4f0ff1e8/lib/remote/createobjecthandler.cpp#L98-L100

> I did it just like in the rest of both the file and lib/remote/.

Might that be the case because because returning an error that way takes around 10 lines instead of 1 to 3 so you're more likely to see that one on first sight? Actually all these functions already have (sometimes even multiple) error conditions that use `HttpUtility::SendJsonError()`:

https://github.com/Icinga/icinga2/blob/e2fda149e536328fa7bbcb62b54725ef4f0ff1e8/lib/remote/createobjecthandler.cpp#L40
https://github.com/Icinga/icinga2/blob/e2fda149e536328fa7bbcb62b54725ef4f0ff1e8/lib/remote/modifyobjecthandler.cpp#L38
https://github.com/Icinga/icinga2/blob/e2fda149e536328fa7bbcb62b54725ef4f0ff1e8/lib/remote/modifyobjecthandler.cpp#L59-L61
https://github.com/Icinga/icinga2/blob/e2fda149e536328fa7bbcb62b54725ef4f0ff1e8/lib/remote/modifyobjecthandler.cpp#L68-L70
https://github.com/Icinga/icinga2/blob/e2fda149e536328fa7bbcb62b54725ef4f0ff1e8/lib/remote/deleteobjecthandler.cpp#L40
https://github.com/Icinga/icinga2/blob/e2fda149e536328fa7bbcb62b54725ef4f0ff1e8/lib/remote/deleteobjecthandler.cpp#L61-L63

And the newly added error condition seems suitable for returning it the same way, so why copy some block of code into 3 places when it can be a simple function call?

Reply:
To harmonise the response structures of a single API endpoint.

Reply:
There is no harmony at the moment, these three endpoints can already return "simple" (as in `HttpUtility::SendJsonError()`) or "complex" (nested `results` and `errors` lists) JSON errors. So when adding a new error condition, why should this use the complex one if the simple one suffices?